collect_app/src/androidTest/java/org/odk/collect/android/support/pages/FormEntryPage.java

```diff
@@ -196,16 +196,19 @@ public FormEntryPage deleteGroup(String questionText) {
     }
 
     public FormEntryPage clickForwardButton() {
+        waitForSoftKeyboardToSettle();
```

Comment:
Could we close the keyboard here instead of using a `Thread.sleep`? I'm also wondering if here's some other condition we could wait for (like what's on the keyboard), but that'd be different for different kinds of questions I guess.

Reply:
No closing the keyboard would also move the buttons and it would also require some delay to wait for this action to finish.

Reply:
Right, but looking at `CloseKeyboardAction`, it seems like it already has code to handle the waiting (with `IdlingResource` it looks like).

Another alternative here would be to use a new custom action that does a `performClick` directly on the view. `ViewActions.click()` sends a tap event to the coordinates of the view (which I guess is the problem here if the view jumps) where as it a direct `performClick()` should still work.

Reply:
Hmm now I'm playing with closing the keyboard again and it seems to work well... This is swear I can swear I've tried it before. Never mind, let's fix it in this way.